docs: Describe how to use Grafana to query logs (#1809)
DSD-DBS/capella-collab-manager: docs/docs/admin/monitoring/logging/explorer.md

<!--
 ~ SPDX-FileCopyrightText: Copyright DB InfraGO AG and contributors
 ~ SPDX-License-Identifier: Apache-2.0
 -->

> # Grafana Explorer

Reviewer: The sidebar name being "Log Explorer" and the title being "Grafana Explorer" is a bit confusing.

Reviewer 2: Changed it to consistent naming: Grafana Log Explorer

The Grafana Explorer can be used to view the logs of different components, such

> as the CCM backend or user sessions. The logs are stored persistently, so the

Reviewer: The abbreviation CCM is only used in one other guide (keycloak), users might not be familiar with it. Consider either expanding the abbreviation on the first usage or just using the full version.

Reviewer 2: Changed it to use the full version.

administrator can also view the logs of already terminated sessions, for
example to investigate an unexpected session termination.

The Grafana Explorer is available to administrators and can be accessed from
the Grafana link in the main menu. Select "Explore" to access the log explorer:

![Explore in the main Grafana menu](./explore.png){:style="width:300px"}

## Select the Correct Source

Before you can view logs, you must make sure that the correct source is
selected for the CCM. For this please select **Loki (Capella Collaboration
Manager)** in the data source picker.

![Select CCM in Data Source Picker](./data_source_picker.png){:style="width:500px"}

## Labels

> The application logs are stored in loki, which uses labels to store and
> retrieve logs efficiently. This means that each log in loki has a set of labels

Reviewer: ```suggestion
The application logs are stored in Loki, which uses labels to store and
retrieve logs efficiently. This means that each log in Loki has a set of labels
```

assigned to it, where a label is essentially a key-value pair.

> The Explorer now consists of two main components, which are described below.

Reviewer: I think you're referring to Label Filters and Operations here but it's not very clear. Consider making the levels of both Component's headings the same, them currently being different is confusing. You might also want to just name the components here.

Reviewer 2: I removed the sentence since I'd neither call "Label filters" nor "Operations" a component.

### Label Filters

> First, we need to select at least one label and a corresponding value to query
> logs. To do this, simply open the **Select Label** drop-down, select a label,

Reviewer: ```suggestion
To query logs, you need to select at least one label and a corresponding value. To do this, simply open the **Select Label** drop-down, select a label,
```
We usually use "we" to refer to the dev team and "you" to refer to the user. I think "you" is clearer here

Reviewer 2: Changed all occurrences to you or, where possible, used the passive form to avoid you / we at all.

provides more structure to the log lines and is especially useful when used
with Loki/Grafana. The reason for this is that we can use the `Logfmt`
operation to convert the key-value pairs in our logs to labels. These labels
can then be used to filter the logs more fine-grained and efficiently.

Here is an example operation sequence where we can query all logs that have the
log level **INFO** and the HTTP method **POST**.

![Operations Example](./operations_example.png){:style="width:500px"}

Theoretically, we could achieve the same result by changing two line contains
operations, but first, this is really inefficient since a string search has to
be applied to all logs, and second, we have to use the exact format as in the
log message.

![Operations Example without Logfmt](./operations_example_bad.png){:style="width:400px"}

## Time Range Selector

Even though Loki is quite efficient, it is still not feasible to always query
all the data, so you need to select the time range for which you want to see
the logs, or use the default "Last 1 hour" time range. To change the time
range, open the drop-down menu in the top right corner and select the time
range you want to see.

> ![Time Range Picker](./time_range.png){:style="width:300px"}

Reviewer: This image looks very small to me, might be different on other screen setups but I would consider making it wider

Reviewer 2: Changed it to 450px
